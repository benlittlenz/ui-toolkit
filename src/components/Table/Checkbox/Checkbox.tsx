import React from 'react';

interface CheckboxProps {
  name: string;
  checked?: boolean;
  onClick?: (e: React.MouseEvent) => void;
}

export function Checkbox({
  name,
  checked = false,
  onClick = () => null,
}: CheckboxProps): JSX.Element {
  return (
    <input
      type="checkbox"
      name={name}
      checked={checked}
      aria-checked={checked}
      onClick={onClick}
      aria-label={name}
      onChange={() => null} // Prevent uncontrolled checkbox warnings
    />
  );
}
