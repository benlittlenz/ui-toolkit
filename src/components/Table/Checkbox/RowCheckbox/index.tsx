import React from 'react';
import styled from 'styled-components';

import { SingleRowAction } from '../../reducers/types';
import { CellBase } from '../../Row/styles';

const TableCellCheckboxStyle = styled(CellBase)`
  flex: 0 0 48px;
  min-width: 48px;
  justify-content: center;
  align-items: center;
  user-select: none;
  white-space: nowrap;
`;

type RowCheckboxProps<T> = {
  name: string;
  keyField: string;
  selected: boolean;
  onSelectedRow: (action: SingleRowAction<T>) => void;
  row: T;
};

export function RowCheckbox<T>({
  name,
  keyField,
  row,
  selected,
  onSelectedRow,
}: RowCheckboxProps<T>): JSX.Element {
  const handleRowSelect = () => {
    onSelectedRow({
      type: 'SELECT_SINGLE_ROW',
      keyField,
      row,
      rowCount: 1,
      isSelected: selected,
      singleSelect: true,
    });
  };
  return (
    <TableCellCheckboxStyle onClick={(e: React.MouseEvent) => e.stopPropagation()}>
      <input
        type="checkbox"
        name={name}
        // checked={selected}
        // aria-checked={selected}
        onClick={handleRowSelect}
        aria-label={name}
      />
    </TableCellCheckboxStyle>
  );
}
