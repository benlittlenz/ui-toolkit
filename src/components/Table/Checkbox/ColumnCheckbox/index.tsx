import React from 'react';
import styled from 'styled-components';

import { AllRowsAction } from '../../reducers/types';
import { CellBase } from '../../Row/styles';
import { Checkbox } from '../Checkbox';

const ColumnStyle = styled(CellBase)`
  flex: 0 0 48px;
  justify-content: center;
  align-items: center;
  user-select: none;
  white-space: nowrap;
  font-size: unset;
`;

interface ColumnCheckboxProps<T> {
  rows: T[];
  selectedRows?: T[];
  allSelected: boolean;
  onSelectAllRows: (action: AllRowsAction<T>) => void;
}

export function ColumnCheckbox<T>({ rows, allSelected, onSelectAllRows }: ColumnCheckboxProps<T>) {
  const rowCount = rows.length;
  const handleSelectAll = () => {
    onSelectAllRows({
      type: 'SELECT_ALL_ROWS',
      rows,
      rowCount,
    });
  };
  return (
    <ColumnStyle onClick={(e: React.MouseEvent) => e.stopPropagation()}>
      <Checkbox
        name="select-all-rows"
        checked={allSelected}
        aria-checked={allSelected}
        onClick={handleSelectAll}
      />
    </ColumnStyle>
  );
}
