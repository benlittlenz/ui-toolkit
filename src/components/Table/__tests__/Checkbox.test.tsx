import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import { Checkbox } from '../Checkbox/Checkbox';

test('it should render the checkbox component and matches the snapshot', () => {
  const { container } = render(<Checkbox name="test" />);

  console.log('container', container);
  expect(container).toMatchSnapshot();
});

test('it should handle onClick', () => {
  const mockClick = jest.fn();
  render(<Checkbox name="test" onClick={mockClick} />);

  userEvent.click(screen.getByRole('checkbox', { name: 'test' }));
  expect(mockClick).toHaveBeenCalled();
});
