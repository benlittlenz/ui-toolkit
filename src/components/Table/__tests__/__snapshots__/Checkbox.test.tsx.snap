// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should handle onClick 1`] = `
<div>
  <input
    aria-checked="false"
    aria-label="test"
    name="test"
    type="checkbox"
  />
</div>
`;

exports[`it should render the checkbox component and matches the snapshot 1`] = `
<div>
  <input
    aria-checked="false"
    aria-label="test"
    name="test"
    type="checkbox"
  />
</div>
`;
